File: backend/src/services/supabaseService.js
const { createClient } = require("@supabase/supabase-js");

const secretKey =
  process.env.SUPABASE_SECRET_KEY || process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!process.env.SUPABASE_URL) {
  throw new Error("SUPABASE_URL is missing");
}

if (!secretKey) {
  throw new Error(
    "SUPABASE_SECRET_KEY or SUPABASE_SERVICE_ROLE_KEY is missing",
  );
}

const supabase = createClient(process.env.SUPABASE_URL, secretKey, {
  auth: {
    persistSession: false,
    autoRefreshToken: false,
    detectSessionInUrl: false,
  },
});

module.exports = supabase;

File: backend/app/services/supabase_service.py
from functools import lru_cache

from fastapi import HTTPException
from supabase import Client, create_client

from app.config import get_settings


def _configured(value: str | None, label: str) -> str:
    if not value:
        raise HTTPException(status_code=500, detail=f"Server configuration error: {label} is missing")
    return value


@lru_cache
def get_supabase() -> Client:
    settings = get_settings()
    return create_client(
        _configured(settings.supabase_url, "SUPABASE_URL"),
        _configured(settings.service_key, "SUPABASE_SECRET_KEY"),
    )


@lru_cache
def get_supabase_auth() -> Client:
    settings = get_settings()
    return create_client(
        _configured(settings.supabase_url, "SUPABASE_URL"),
        _configured(settings.public_key, "SUPABASE_PUBLISHABLE_KEY"),
    )
